### ISSUE TITLE
Default Consolidation Period Is Too Short

### ISSUE THREAD

**Issue by richardjennings-fc** [aws/karpenter#2606]
### Version

__Karpenter Version:__ v0.0.0

__Kubernetes Version:__ v1.0.0

### Expected Behavior

A reasonable default configuration balancing stability with efficiency.

### Actual Behavior

The default consolidation interval of 10 seconds is too short for some classes of workload.

For environments with Pods that take a long time to startup, the disruption caused by the frequent controller loop is problematic to stability.

The value I believe should be set higher in the general case.

Specifically I believe there is no value which is correct for all cases and as such it should be made configurable.

### Steps to Reproduce the Problem

Run a cluster with Pods that are slow to start and observe the consolidation interval resulting in churn and instabilities.

### Resource Specs and Logs

n/a

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by tzneal** [comment]
Can you give an example that reproduces this?  Specifically we check if the deployment is [reporting that it has the desired number of ready replicas](https://github.com/aws/karpenter/blob/077c099b4419d0caac447904d9ffae1d871e37b9/pkg/controllers/consolidation/controller.go#L592) before consolidation occurs.

**Comment by richardjennings-fc** [comment]
> Can you give an example that reproduces this? Specifically we check if the deployment is [reporting that it has the desired number of ready replicas](https://github.com/aws/karpenter/blob/077c099b4419d0caac447904d9ffae1d871e37b9/pkg/controllers/consolidation/controller.go#L592) before consolidation occurs.

This is true - but does not prevent the deployment from being disrupted 10 seconds after it becomes ready. If the [Consolidation Polling Period](https://github.com/aws/karpenter/blob/main/pkg/controllers/consolidation/controller.go#L61) was longer, the deployment can expect a longer period of time before potentially being disturbed by consolidation. Preferably the Polling Period would be tuneable to match the requirements of specific workloads and environments.

**Comment by richardjennings-fc** [comment]
Pods that are Ready, in Deployments that are Available - can be disturbed by Consolidation potentially every 10 second Polling Period. For some workloads this is too frequent.

**Comment by tzneal** [comment]
Are there PDBs on the pods? That will throttle the rate at which they are evicted which will also throttle consolidation.

**Comment by tzneal** [comment]
@richardjennings-fc If this is an issue, there's no need to close it.  I just want to work out the root cause and determine if the existing mechanisms for throttling consolidation are sufficient.  If not, feel free to re-open.

**Comment by richardjennings-fc** [comment]
@tzneal After consolidation was turned off we were able to identify that a newly added component was launching pods expected to behave like jobs. Karpenter would correctly attempt to consolidate these pods which due to some design decision that might not make sense cannot be interrupted. We will remove the immediate issue with annotations to prevent consolidation for these strange pods. I still believe that the consolidation loop frequency should be configurable.
